src/Orc.FilterBuilder/Orc.FilterBuilder.Shared/Models/FilterScheme.cs

```diff
@@ -99,8 +99,9 @@ private void OnConditionItemsChanged()
             SubscribeToEvents();
         }
 
-        protected override void OnDeserialized()
+        protected async override void OnDeserialized()
         {
+            await this.EnsureIntegrityAsync();
```

Comment:
Not sure why you picked the async version inside a non-async method? Can you explain?